govcd/edgegateway.go

```diff
@@ -1076,3 +1076,88 @@ func (eGW *EdgeGateway) buildProxiedEdgeEndpointURL(optionalSuffix string) (stri
 
 	return hostname, nil
 }
+
+// GetLoadBalancer retrieves load balancer configuration of `&types.LoadBalancer` and can be used
+// to access global configuration options. These are 4 fields only:
+// LoadBalancer.Enabled, LoadBalancer.AccelerationEnabled, LoadBalancer.Logging.Enable,
+// LoadBalancer.Logging.LogLevel
+func (egw *EdgeGateway) GetLoadBalancer() (*types.LoadBalancer, error) {
+	if !egw.HasAdvancedNetworking() {
+		return nil, fmt.Errorf("only advanced edge gateway supports load balancing")
+	}
+
+	httpPath, err := egw.buildProxiedEdgeEndpointURL(types.LBConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	loadBalancerConfig := &types.LoadBalancer{}
+	_, err = egw.client.ExecuteRequest(httpPath, http.MethodGet, types.AnyXMLMime,
+		"unable to read load balancer configuration: %s", nil, loadBalancerConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return loadBalancerConfig, nil
+}
+
+// UpdateLoadBalancer allows to update global load balancer configuration.
+// These are 4 fields only: LoadBalancer.Enabled, LoadBalancer.AccelerationEnabled,
+// LoadBalancer.Logging.Enable, LoadBalancer.Logging.LogLevel. They are represented in load balancer
+// global configuration tab in the UI.
+// All other fields are ignored and sent as they are in order to prevent load balancer configuration
+// corruption
+func (egw *EdgeGateway) UpdateLoadBalancer(lb *types.LoadBalancer) (*types.LoadBalancer, error) {
```

Comment:
Someone may trip on the function name. What about `UpdateLoadBalancerGlobal` to be more specific?

Reply:
Global? - I think it's wrong. It's only for 4 fields. We need to be more specific. I agree current name isn't enough and we need other ones `EnableLoadBalancer`, `UpdateLoadBalancerEnalblament`.. other options? UpdateGenericSomething should be used for main update functionality.

Reply:
Changed.

Reply:
I must admit that `UpdateLoadBalancerGlobal` doesn't sound to precise as well as `UpdateLoadBalancer`, but saying that it is only `enable` is confusing as you can also tune logging options in this tab as per screenshot below. The `global` comes from tab name, but not sure what other names we could make:
* `UpdateLoadBalancerConfig` ?

If I split this to only `enable` then there will be more functions for logging and more calls which I'd say are not necessary.



<img width="716" alt="image" src="https://user-images.githubusercontent.com/15804230/61707524-5adf8680-ad53-11e9-9519-1497c5120310.png">

Reply:
I agree enable is just partly say what is doing. Definitely against *Global. Also don't like *Config in this case at this also very abstract. We need brainstorm session :)

Reply:
I saw that UI names this Global configuration, but from API point of view is just flat fields of LB

Reply:
Then, at least to me looks like `UpdateLoadBalancerGlobalConfig` would be most precise even though long. What do you say?

Reply:
It possible, but I don't like it, cause it's UI name, not API. We agree that govcd matches API and terraform UI namings.

Reply:
We found that NSX API calls it "General Load Balancer Parameters"
![image (1)](https://user-images.githubusercontent.com/11330689/61719036-d863c080-ad6c-11e9-89c4-d66060e54f34.png)

Reply:
Decided:
```
UpdateLBGeneralParams(<four fields>)
GetLBGeneralParams() (*types.LBGeneralParamsWithXML, error)
```